app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = dict()
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    res_list = []
+
+    for dic in people:
+        if None in dic.values():
+            res_list.append(Person(dic["name"], dic["age"]))
+        elif "wife" in dic.keys():
+            person = Person(dic["name"], dic["age"])
+            person.wife = dic["wife"]
+            res_list.append(person)
+        elif "husband" in dic.keys():
+            person = Person(dic["name"], dic["age"])
+            person.husband = dic["husband"]
+            res_list.append(person)
```

Comment:
There is no need for a two-time search for wife and husband. By the first iteration of the people list need to create an object of class without a partner and by the second iteration add them

Reply:
optimize done